client/components/infinite-scroll/index.js

```diff
@@ -3,53 +3,92 @@
 /**
  * External dependencies
  */
-import { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { throttle } from 'lodash';
+import { throttle, defer } from 'lodash';
+import afterLayoutFlush from 'lib/after-layout-flush';
 
 const SCROLL_CHECK_RATE_IN_MS = 400;
 
-export default class InfiniteScroll extends Component {
+export default class InfiniteScroll extends React.Component {
 	static propTypes = {
 		nextPageMethod: PropTypes.func.isRequired,
 	};
 
-	checkScrollPositionHandler = throttle(
-		() => this.checkScrollPosition( true ),
-		SCROLL_CHECK_RATE_IN_MS
-	);
+	observedElement = React.createRef();
 
 	componentDidMount() {
-		window.addEventListener( 'scroll', this.checkScrollPositionHandler );
-		this.checkScrollPosition( false );
+		this.hasIntersectionObserver = 'IntersectionObserver' in window;
+
+		if ( this.hasIntersectionObserver && this.observedElement.current ) {
+			this.observer = new IntersectionObserver( this.handleIntersection, {
+				rootMargin: '100%',
+				threshold: 1.0,
+			} );
+			this.observer.observe( this.observedElement.current );
+		} else {
+			window.addEventListener( 'scroll', this.checkScrollPositionHandler );
+			this.deferredTimer = defer( () => this.checkScrollPositionHandler( false ) );
```

Comment:
With `throttle` and `afterLayoutFlush`, this `defer` call becomes unneccessary. `afterLayoutFlush` itself schedules a rAF followed by `setTimeout( 0 )`. So, with `defer`, the chain becomes:
```js
setTimeout( // from defer
  () => requestAnimationFrame( // from afterLayoutFlush
    () => setTimeout( // fromAfterLayoutFlush
      checkScroll,
      0
    )
  ),
  0
);
```
See the extra `setTimeout( 0 )`? 

Fun fact: with `const throttled = throttle( check, 400 )`, the first call to `throttled` calls `check` synchronously, only the subsequent calls withing the throttling period are scheduled for later.

Reply:
Thank you for clearing all of that up! Yes, it does seem I was over-scheduling. Fixed!